### DISCUSSION TITLE
multi environment monorepo ignore build step not working for commit

### DISCUSSION THREAD

**Discussion by codercatdev** [vercel/vercel#6935] [Help]
I am trying to get only a specific build from each folder to trigger when using the git integration. I have this in each of my folders and point two separate projects. One for Next.js and one for Sanity Studio. 

My goal is to only have this build when each is affected by using the script below. But it still seems to build each.

build-dev.sh
```
#!/bin/bash

result=$(git diff HEAD^ HEAD --quiet .)

if [[ $result == true ]]; then
  echo "🛑 - Build other folder"
  exit 0;
else
  echo "✅ - Build this folder"
fi

echo "VERCEL_GIT_COMMIT_REF: $VERCEL_GIT_COMMIT_REF"

if [[ "$VERCEL_GIT_COMMIT_REF" != "stage" && "$VERCEL_GIT_COMMIT_REF" != "main"  ]] ; then
  # Proceed with the build
    echo "✅ - Build can proceed"
  exit 1;

else
  # Don't build
  echo "🛑 - Build cancelled"
  exit 0;
fi
```

## Folder structure

![image](https://user-images.githubusercontent.com/3102249/139951603-62eabebb-4030-452b-8592-a777a81efebd.png)

## Environments
![image](https://user-images.githubusercontent.com/3102249/139951722-8c78ee9c-c1f4-45d5-b3c9-9e0b4a85c319.png)

## Root directories
`backend/sanity`
![image](https://user-images.githubusercontent.com/3102249/139951781-d63dc84b-873e-4c2a-89dc-0445f9b7b97a.png)
`frontend/main`
![image](https://user-images.githubusercontent.com/3102249/139951811-2134930c-4d20-4fbd-82b4-1b998d53590e.png)

## Latest Change Example

This only has a change in `frontend/main/src/pages/api/algolia.ts`
https://github.com/CodingCatDev/codingcat.dev/commit/fefe37c1e0159ece77623f8bc4f757beb9f58efd

### Next.js dev-codingcat-dev
![image](https://user-images.githubusercontent.com/3102249/139952122-5ae68230-d507-46f8-b248-b87a84d8535f.png)

### Sanity admin-dev-codingcat-dev
![image](https://user-images.githubusercontent.com/3102249/139952261-135ee00f-d57d-4367-8e25-38498e4dfa22.png)

An example of the second step working great
```
17:05:38.667 | Cloning github.com/CodingCatDev/codingcat.dev (Branch: feature/sanity-live-update, Commit: fefe37c)
-- | --
17:05:39.591 | Cloning completed: 923.568ms
17:05:39.591 | Running "bash build-main.sh"
17:05:39.600 | ✅ - Build this folder
17:05:39.601 | VERCEL_GIT_COMMIT_REF: feature/sanity-live-update
17:05:39.601 | 🛑 - Build cancelled
17:05:39.602 | The Deployment has been canceled as a result of running the command defined in the "Ignored Build Step" setting.
```

**Comment by leerob** [comment]
Hey! Thank you for providing all of this information. We're working on improving our monorepo support further and can report back here when these new improvements land.
